common/models/user.js

```diff
@@ -580,7 +580,10 @@ module.exports = function(User) {
             query && query.redirect !== undefined ? query.redirect :
             undefined;
 
-          ctx.res.redirect(redirectUrl);
+          if (redirectUrl !== undefined) {
+            ctx.res.location(redirectUrl);
+          }
+          next()
         } else {
           next(new Error('transport unsupported'));
```

Comment:
The error should be reported only when there is `redirectUrl`. When the url is not provided then there is no reason to throw the error.